stencil-library/src/components/payment-details/payment-details.tsx

```diff
@@ -52,31 +52,38 @@ export class PaymentDetails {
   }
 
   async fetchData(): Promise<void> {
-    this.errorMessage = '';
     if (!this.paymentId || !this.authToken) {
       this.errorMessage = "Can not fetch any data without a PaymentID and an AuthToken";
       this.loading = false;
       return;
     }
     this.loading = true;
-    const endpoint = `payments/${this.paymentId}`;
-    const response: IApiResponse<IPayment> = await Api(this.authToken, config.proxyApiOrigin).get(endpoint);
 
-    if (!response.error) {
-      this.payment = new Payment(response.data);
-    } else {
-      this.errorMessage = typeof response.error === 'string' ? response.error : response.error.message;
-    }
+    try {
+      const endpoint = `payments/${this.paymentId}`;
+      const response: IApiResponse<Payment> = await Api(this.authToken, config.proxyApiOrigin).get(endpoint);
 
-    this.loading = false;
+      if (!response.error) {
+        this.payment = response.data;
+      } else {
+        const responseError = typeof response.error === 'string' ? response.error : response.error.message;
+        this.errorMessage = `Error trying to fetch data : ${responseError}`;
+        console.error(this.errorMessage);
+      }
+    } catch (error) {
+      this.errorMessage = `Error trying to fetch data : ${error}`;
+      console.error(this.errorMessage);
+    } finally {
+      this.loading = false;
+    }
   }
 
   render() {
+    if (this.loading) return LoadingState;
+    if (this.errorMessage) return ErrorState(this.errorMessage);
     return (
       <Host>
-        {this.loading && LoadingState}
-        {!this.loading && this.errorMessage && ErrorState(this.errorMessage)}
-        {!this.loading && this.payment && (
+        {this.payment && (
```

Comment:
I feel like `LoadingState` and `ErrorState` should both be rendered inside the `Host` tag? 🤔 

```
<Host>
        {this.loading && LoadingState}
        {!this.loading && this.errorMessage && ErrorState(this.errorMessage)}
        {!this.loading && this.payment && 
             <justifi-details ....... />
</Host>
```

Reply:
Yes, I was thinking in checking loading and error states first and then returning, to avoid checkings like `!this.loading && this.errorMessage` but forgot about the <Host> tag.

I am putting them back